### PR TITLE
Key signature and wallet address get functions added to Public key

### PR DESCRIPTION
**Pull request by aprasolova**
### Problem

Signature and wallet address are entities derived from NEO public key, but regarding functions are defined on Private key only. Thus, in case when I have a public key and cannot obtain private key, I cannot obtain signature/address as well, despite I have no conceptual restrictions to do it.

### Solution

Define key signature calculation and wallet address conversion functions on Public key entity.

### CODE REVIEW

File: pkg/wallet/private_key.go [hunk 1]
```diff
@@ -93,7 +93,12 @@ func (p *PrivateKey) PublicKey() ([]byte, error) {
 	}
 	b := append(prefix, padded...)
 
-	return b, nil
+	pk := &crypto.PublicKey{}
```
Comment (im-kulikov) [inline pkg/wallet/private_key.go]: move to `var`, remove `&`

Comment (im-kulikov) [reply pkg/wallet/private_key.go]:
```go
var (
// ...
pk crypto.PublicKey
)
// ...
return &pk, nil
```

**Comment by hal0x2328** [comment]
Award aprasolova 200 points
Award im-kulikov 50 points for review

**Comment by lllwvlvwlll** [comment]
@aprasolova please provide an address for awards.

**Comment by aprasolova** [comment]
@lllwvlvwlll  APBNgjMFPsBywPc34FTLF2ciR9jxnb7ajJ

**Comment by lllwvlvwlll** [comment]
https://neoscan.io/transaction/28324feb9528c493f6588f8a7a6d559820e01581b82f7fd63c5b52a344333d52

**Pull request merged**